### PR TITLE
feat: Edit report name - Spender changes

### PR DESCRIPTION
**Pull request by Dimple16**
### Changes
- Spender can edit the report purpose in the report states up to Approval pending state.
- All submitted reports will have option to edit report name in the individual view report page
- Platform reports `POST` API is used to edit the report purpose
- Code coverage is added for all the new methods
- Models are added for all the new objects

### <samp>🤖 Generated by Copilot at 91d0af5</samp>

This pull request adds a feature to allow users to edit the report purpose in the my view report page. It also adds and updates the models, mock data and tests for the report service and the my view report page. It modifies the report service to use the platform report model and to add a method for updating the report purpose. It updates the my view report page template and file to use the new report service method and to show the edit icon for the report purpose in more states.

### <samp>🤖 Generated by Copilot at 91d0af5</samp>

> _We mock the data for the platform report_
> _We define the models and the types_
> _We edit the purpose of our fate_
> _We update the name with `updateReportName`_

### <samp>🤖 Generated by Copilot at 91d0af5</samp>

*  Add mock data, model files and test cases for platform report and report approvals ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-cf19e20099d68d312b53171ad025bc92dee77b66f8da7bfe8f8ce42f8500ad7aR1-R64), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-8ac4fa0d808c14c87b412180a9b3e54c654818092676ee68cebc18c304fe4d95R855-R909), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-a52c023c82f91743cd908d3579a8f5ccef6a1172ca3120cc44ad1ce0ed61fcf5R1-R63), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-fd004e1ba7ba09eb5279840c250e2041a57c8bbc8df02bacab51c3f9e5bef5ebR1-R9), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-609c1fb70188be62ef86ff68c50f64df0f710b41ce36c8b1109fc2034ee03524R55), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-609c1fb70188be62ef86ff68c50f64df0f710b41ce36c8b1109fc2034ee03524R73), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-609c1fb70188be62ef86ff68c50f64df0f710b41ce36c8b1109fc2034ee03524R686-R700), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-fe3424c28721c88278049c01f15468c2e33fa02bc82417ae132a2a7c5ae977bcR59))
*  Import platform report model file in report service and my view report page files ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9R35), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-fe3424c28721c88278049c01f15468c2e33fa02bc82417ae132a2a7c5ae977bcR59))
*  Implement updateReportPurpose method in report service to call spenderPlatformV1ApiService with report id, source and purpose fields and return platform report observable ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9R239-R252))
*  Modify updateReportName method in my view report page to call updateReportPurpose method from report service and return platform report observable ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL310-R310), [link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-fe3424c28721c88278049c01f15468c2e33fa02bc82417ae132a2a7c5ae977bcL553-R559))
*  Add updateReportPurpose method to spy list in my view report page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-fe3424c28721c88278049c01f15468c2e33fa02bc82417ae132a2a7c5ae977bcR87))
*  Modify condition for showing edit icon for report purpose in my view report page template to include APPROVER_PENDING state ([link](https://github.com/fylein/fyle-mobile-app/pull/2244/files?diff=unified&w=0#diff-a52aaafa54cc049d42c344ff3d466b8f46ad3b7aca50bda871b8f6258071f3d4L38-R38))

## Clickup
https://app.clickup.com/t/85ztqzfxx

### Report Service changes coverage - 100%
Screenshot 2023-08-09 at 7 44 10 PM

### My view report page coverage - 100%
Screenshot 2023-08-09 at 7 45 02 PM

## UI Preview
Screenshot 2023-08-09 at 6 32 19 PM

Screenshot 2023-08-09 at 6 32 33 PM

Screenshot 2023-08-09 at 6 32 40 PM

Screenshot 2023-08-09 at 6 32 56 PM

Screenshot 2023-08-09 at 6 33 40 PM

Screenshot 2023-08-09 at 6 33 51 PM

### CODE REVIEW

File: src/app/core/services/report.service.spec.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ import {
   expectedReportSingleResponse,
   reportParam,
   expectedPaginatedReports,
+  reportData1,
```
Comment (abhishek1234321) [inline src/app/core/services/report.service.spec.ts:55]: why is named like this? `reportData1` I mean

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/core/mock-data/platform-report.data.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { PlatformReport } from '../models/platform/platform-report.model';
+
+export const platformReportData: PlatformReport = {
+  amount: 0,
+  approvals: [],
+  created_at: new Date('2023-07-11T06:19:28.260142+00:00'),
+  currency: 'INR',
```
Comment (rvab) [inline src/app/core/mock-data/platform-report.data.ts:7]: use `USD`

File: src/app/core/models/platform/report-approvals.model.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface ReportApprovals {
+  approver_user_id: string;
+  approver_user: {
```
Comment (rvab) [inline src/app/core/models/platform/report-approvals.model.ts:3]: Why not use Pick here? Pick<User, 'id' | 'email' | 'full_name'>

**Review by rvab — APPROVED** [review]
minor comments

File: src/app/core/services/report.service.spec.ts (hunk 1 above)
Comment (Dimple16) [reply src/app/core/services/report.service.spec.ts:55]: in the mock data file we have multiple data with the same model

**Pull request merged**